x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -56,6 +56,20 @@ public void process(NativePyTorchProcess process) {
             if (isStopping == false) {
                 logger.error(new ParameterizedMessage("[{}] Error processing results", deploymentId), e);
             }
+            // If we are stopping, or the process is no longer alive, notify all the pendingResults listeners
+            // Otherwise, we don't particularly know which result failed
+            if (isStopping || (process.isProcessAliveAfterWaiting() == false)) {
```

Comment:
Once execution reaches this point in the exception handler there is no going back. We should call all the pending handlers regardless of the status as they will just timeout otherwise.

Reply:
> Once execution reaches this point in the exception handler there is no going back. We should call all the pending handlers regardless of the status as they will just timeout otherwise.

@davidkyle so, this exception catch will cause ALL pending requests to fail? That doesn't really reflect the current behavior. If that were the case, we should just crash the entire process if an exception is handled. But, I don't think we should do that.

Reply:
This is after the `while (iterator.hasNext()) {` loop so no  further results will be processed even if the native process is still writing them. Any pending requests that have not had their latch counted down will fail now as they will never get the result. We might as well call the listeners here with this exception instead of letting them timeout. 

Once this function exits and the result processing thread exits the only thing to do is stop and restart the deployment if you want more results.

Reply:
Ah, I didn't fully grok what `while (iterator.hasNext()) {` was doing.